webtestingii/src/App.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from "react";
+import Display from "./Display";
+import Dashboard from "./Dashboard";
+import "./App.css";
+
+class App extends Component {
+  state = {
+    strikes: 0,
+    balls: 0,
+    outs: 0,
+    runs: 0
+  };
+
+  baseHit = () => {
+    this.setState(prevState => ({
+      balls: 0,
+      strikes: 0,
+      runs: prevState.runs + 1
+    }));
+  };
+
+  foulBall = () => {
+    this.setState(prevState => ({
+      strikes: prevState.strikes < 2 ? prevState.strikes + 1 : prevState.strikes
+    }));
+  };
+
+  strike = () => {
+    this.setState(prevState => ({
+      balls: prevState.strikes === 2 ? 0 : prevState.balls,
+      strikes: prevState.strikes === 2 ? 0 : prevState.strikes + 1,
+      outs: prevState.strikes === 2 ? prevState.outs + 1 : prevState.outs
+    }));
```

Comment:
you could cut down a little and make you count reset DRY by creating a function that resets the state of balls and strikes, then just invoke it here if strikes gets to 3 or in `ball` if that ever gets to 4